receiver/routes/adguard.py

```
@@ -0,0 +1,172 @@
+        )
+
+    set_config(enricher_db, 'adguard_enabled',      body.enabled)
+    set_config(enricher_db, 'adguard_host',         body.host.rstrip('/'))
+    set_config(enricher_db, 'adguard_username',     body.username)
+    set_config(enricher_db, 'adguard_poll_interval', body.poll_interval)
+
+    if body.password and body.password != _PASSWORD_PLACEHOLDER:
+        try:
+            set_config(enricher_db, 'adguard_password_enc', encrypt_api_key(body.password))
+        except ValueError as e:
+            raise HTTPException(status_code=500, detail=f'Encryption failed: {e}')
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)